app/controllers/stories_controller.rb

```diff
@@ -168,13 +168,11 @@ def handle_page_display
 
   def handle_base_index
     @home_page = true
-    assign_feed_stories
+    assign_feed_stories unless user_signed_in? # Feed fetched async for signed-in users
```

Comment:
This is only needed when the user is _not_ signed in, and because this ultimately turns into an array and sorts, it will be queried even if we don't use it on the view. This path is mostly cached so it wasn't causing us huge problems, but one less code path traversed on signed-in cold hits. This should reduce app memory consumption by a bit.

I consolidated all the logic related to this.

Reply:
Would be great to have it summarized in a comment